backend/src/controllers/groups-controller.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from 'express';
+import { v4 as uuid } from 'uuid';
+
+interface Group {
+    name:string;
+    _id:string;
+}
+
+let groups: Group[] = []; //Array of groups
+
+//Creates a group with the desired name and gives a unique ID
+export const createGroup = async (req: Request, res: Response) => {
+    const { name }: { name: string } = req.body;
+
+    //Checks if there's a similar group
+    if (groups.find((group) => group.name === name)) {
+        return res.status(409).json({ error: "The group name was already taken"});
+    }
+    //Validates the provided name
+    if (typeof name !== 'string' || name.length > 30 ){
```

Comment:
Use `express-validator` in the routes file to ensure values in the body are correctly formatted